cgsmiles/drawing_utils.py

```diff
@@ -0,0 +1,193 @@
+from collections import defaultdict
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+
+def angle_between(v1, v2):
+    cos_theta = np.dot(v1, v2) / (np.linalg.norm(v1) * np.linalg.norm(v2))
+    angle_rad = np.arccos(np.clip(cos_theta, -1.0, 1.0))
+    return np.degrees(angle_rad)
+
+def rotation_from_x_axis(v):
+    angle_rad = np.arctan2(v[1], v[0])
+    return np.degrees(angle_rad)
+
+def angle_of_interest(v1, v2):
+    vtot = v1 + v2
+    # Rotation angle with respect to the x-axis
+    angle_x_axis = rotation_from_x_axis(vtot)
+    return angle_x_axis
+
+def create_rwb_colormap(n):
+    cmap = plt.get_cmap('tab20', n)
+    colors = cmap(np.linspace(0, 1, n))
+    return ListedColormap(colors)
+
+default_colormap = create_rwb_colormap
```

Comment:
https://matplotlib.org/stable/users/explain/colors/colormaps.html#qualitative
If you look at the tab20 cmap, you kind of want to first pick even indexed colors, then odd indexed (i.e. dark blue, dark orange, ..., dark cyan, light blue, ..., light cyan), so I'm not convinced remapping it to `n<20` elements is what you want to do.